File: scripts/reference-solutions/non-overlapping-intervals.cpp
#include <algorithm>
#include <iostream>
#include <utility>
#include <vector>

using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  int n;
  cin >> n;

  vector<pair<long long, long long>> intervals(n);
  for (auto& interval : intervals) {
    cin >> interval.first >> interval.second;
  }

  sort(intervals.begin(), intervals.end(), [](const auto& left, const auto& right) {
    if (left.second != right.second) {
      return left.second < right.second;
    }
    return left.first < right.first;
  });

  int removed = 0;
  bool hasCurrent = false;
  long long currentEnd = 0;

  for (const auto& [start, end] : intervals) {
    if (!hasCurrent || start >= currentEnd) {
      currentEnd = end;
      hasCurrent = true;
    } else {
      ++removed;
    }
  }

  cout << removed << '\n';
  return 0;
}

File: scripts/reference-solutions/non-overlapping-intervals.py
import sys

data = sys.stdin.read().strip().split()
n = int(data[0])
intervals = []
index = 1
for _ in range(n):
    start = int(data[index])
    end = int(data[index + 1])
    index += 2
    intervals.append((start, end))

intervals.sort(key=lambda interval: (interval[1], interval[0]))
removed = 0
current_end = None

for start, end in intervals:
    if current_end is None or start >= current_end:
        current_end = end
    else:
        removed += 1

print(removed)
